util/include/crypto/eddsa/EdDSAVerifier.hpp

```diff
@@ -11,42 +11,60 @@
 // LICENSE file.
 //
 #pragma once
-#include "EdDSA.h"
+
+#include "EdDSA.hpp"
 #include "openssl_crypto.hpp"
+#include "crypto_utils.hpp"
+#include "crypto/interface/verifier.hpp"
+
+// OpenSSL includes.
+#include <openssl/pem.h>
+
+namespace concord::crypto::openssl::eddsa {
 
 /**
- * @tparam PublicKeyType The type of the public key - expected to be a SerializableByteArray
- *
+ * @tparam PublicKeyType The type of the public key, expected to be a SerializableByteArray.
  */
 template <typename PublicKeyType>
-class EdDSAVerifier {
+class EdDSAVerifier : public IVerifier {
  public:
   using VerifierKeyType = PublicKeyType;
-  EdDSAVerifier(const PublicKeyType &publicKey) : publicKey_(publicKey) {}
 
-  bool verify(const std::string &message, const std::string &signature) const {
+  /**
+   * @brief Construct a new EdDSA Verifier object
+   *
+   * @param publicKey
+   */
+  explicit EdDSAVerifier(const PublicKeyType &publicKey) : publicKey_(publicKey) {}
+
+  bool verify(const std::string &message, const std::string &signature) const override {
     return verify(message.data(), message.size(), signature.data(), signature.size());
   }
 
-  bool verify(const char *msg, int msgLen, const char *sig, int sigLen) const {
-    return verify(reinterpret_cast<const uint8_t *>(msg),
-                  static_cast<size_t>(msgLen),
-                  reinterpret_cast<const uint8_t *>(sig),
-                  static_cast<size_t>(sigLen));
+  bool verify(const char *msg, std::size_t msgLen, const char *sig, std::size_t sigLen) const {
+    return verify(reinterpret_cast<const uint8_t *>(msg), msgLen, reinterpret_cast<const uint8_t *>(sig), sigLen);
   }
 
   bool verify(const uint8_t *msg, size_t msgLen, const uint8_t *sig, size_t sigLen) const {
+    using concord::util::openssl_utils::UniquePKEY;
```

Comment:
change the namesopace name to concord::crypto::openssl

Reply:
Changed in https://github.com/arc-vmware/concord-bft/pull/6/commits/87f4edd0f4f5d72b422cef50f4a8eeeee8b28bc2